coderd/database/dbauthz/dbauthz.go

```diff
@@ -1166,13 +1214,28 @@ func (q *querier) GetOAuth2ProviderAppSecretsByAppID(ctx context.Context, appID
 	return q.db.GetOAuth2ProviderAppSecretsByAppID(ctx, appID)
 }
 
+func (q *querier) GetOAuth2ProviderAppTokenByAppSecretIDAndSecret(ctx context.Context, arg database.GetOAuth2ProviderAppTokenByAppSecretIDAndSecretParams) (database.OAuth2ProviderAppToken, error) {
+	if err := q.authorizeContext(ctx, rbac.ActionRead, rbac.ResourceOAuth2ProviderAppCodeToken); err != nil {
+		return database.OAuth2ProviderAppToken{}, err
+	}
+	return q.db.GetOAuth2ProviderAppTokenByAppSecretIDAndSecret(ctx, arg)
+}
+
```

Comment:
Can we just add `UserID` to `database.OAuth2ProviderAppToken{}` to make the RBAC easier? Currently you can only read this if you can read all tokens? Like an owner?

Should it not be `rbac.ResourceOAuth2ProviderAppCodeToken.WithOwner(userID)`?

Reply:
Also make a `RBACObject` method for it.

Reply:
Huh apparently I am not actually using this anywhere so I suppose I can just delete this for now.  I am not sure what my original intention with it was.